backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/process/C_BPartner_AddRemoveCreditStopStatus.java

```diff
@@ -41,17 +39,21 @@ protected String doIt()
 
 		final BPartnerStats stats = bpartnerStatsDAO.getCreateBPartnerStats(bPartner);
 		final String creditStatus;
-		if (isSetCreditStop)
+
+		if (SetCreditStatusEnum.CreditOK.equals(setCreditStatus)) {
+			creditStatus = X_C_BPartner_Stats.SOCREDITSTATUS_CreditOK;
+		}
+		else if (SetCreditStatusEnum.CreditStop.equals(setCreditStatus))
 		{
 			creditStatus = X_C_BPartner_Stats.SOCREDITSTATUS_CreditStop;
 		}
 		else
 		{
 			final CalculateSOCreditStatusRequest request = CalculateSOCreditStatusRequest.builder()
```

Comment:
now I saw this :(
basically, it's not a good practice to not cover all values of an enum.

so basically I would do 
else if(SetCreditStatusEnum.Calculate.equals(setCreditStatus)) {
// do the calculation
}
else {
throw new AdempiereException("Invalid setCreditStatus value: " + setCreditStatus);
}

Reply:
done